src/http-gateways/path-gateway.md

```diff
@@ -359,23 +373,46 @@ See: [Denylists](#denylists)
 
 ### `500` Internal Server Error
 
-A generic server error returned when it is not possible to return a better one.
+A generic server error returned when it is not possible to return a better
+one. An internal server error signals the general unavailability of the
+gateway.
 
 ### `502` Bad Gateway
 
-Returned immediately when Gateway was not able to produce response for a known reason.
-For example, when gateway failed to find any providers for requested data.
+Error that indicates that a Gateway was not able to produce response for a
+known reason: for example, in the case of
+[recursive gateways](#recursive-vs-non-recursive-gateways), in the event of
+failure to find any providers for requested data. `502` indicates that the
+request can be retried and is not a permanent failure.
 
-This error response SHOULD include [`Retry-After`](#retry-after-response-header) HTTP header to indicate how long the client should wait before retrying.
+This error response SHOULD include
+[`Retry-After`](#retry-after-response-header) HTTP header to indicate how long
+the client should wait before retrying.
+
+Gateways MUST return `404` instead of `502` when the content is known to be
+unretrivable: for example, when the Gateway is
+[non-recursive](#recursive-vs-non-recursive-gateways) and the content is known
+to not be available.
 
 ### `504` Gateway Timeout
 
-Returned when Gateway was not able to produce response under set time limits.
-For example, when gateway failed to retrieve data from a remote provider.
+Error that indicates that the Gateway was not able to produce response under
+set time limits: for example, when gateway failed to retrieve data from a
+remote provider. `504` indicates that the request can be retried and is not a
+permanent failure.
+
+There is no generic timeout, Gateway implementations SHOULD set timeouts based
+on specific use cases.
+
+This error response SHOULD include
+[`Retry-After`](#retry-after-response-header) HTTP header to indicate how long
+the client should wait before retrying.
 
-There is no generic timeout, Gateway implementations SHOULD set timeouts based on specific use cases.
+Gateways MUST return `404` instead of `504` when the content is known to be
+unretrivable: for example, when the Gateway is
+[non-recursive](#recursive-vs-non-recursive-gateways) and the content is known
+to not be available.
```

Comment:
```suggestion
Gateways SHOULD return `404` instead of `504` when the content is known to be
unretrievable: for example, when the Gateway is
[non-recursive](#recursive-vs-non-recursive-gateways) and the content is known
to not be available.
```